quickwit-metastore/src/backward_compatibility_tests/index_metadata.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::BTreeSet;
+
+use byte_unit::Byte;
+use quickwit_config::{
+    DocMapping, IndexingResources, IndexingSettings, MergePolicy, SearchSettings, SourceConfig,
+};
+use quickwit_index_config::SortOrder;
+
+use crate::checkpoint::{Checkpoint, CheckpointDelta, PartitionId, Position};
+use crate::IndexMetadata;
+
+pub(crate) fn test_index_metadata_eq(
+    index_metadata: &IndexMetadata,
+    expected_index_metadata: &IndexMetadata,
+) {
+    assert_eq!(index_metadata.index_id, expected_index_metadata.index_id);
+    assert_eq!(index_metadata.index_uri, expected_index_metadata.index_uri);
+    assert_eq!(
+        index_metadata.checkpoint,
+        expected_index_metadata.checkpoint
+    );
+    assert_eq!(
+        index_metadata
+            .doc_mapping
+            .field_mappings
+            .iter()
+            .map(|field_mapping| &field_mapping.name)
+            .collect::<Vec<_>>(),
+        expected_index_metadata
+            .doc_mapping
+            .field_mappings
+            .iter()
+            .map(|field_mapping| &field_mapping.name)
+            .collect::<Vec<_>>(),
+    );
+    assert_eq!(
+        index_metadata.doc_mapping.tag_fields,
+        expected_index_metadata.doc_mapping.tag_fields,
+    );
+    assert_eq!(
+        index_metadata.doc_mapping.store_source,
+        expected_index_metadata.doc_mapping.store_source,
+    );
+    assert_eq!(
+        index_metadata.indexing_settings,
+        expected_index_metadata.indexing_settings
+    );
+    assert_eq!(
+        index_metadata.search_settings,
+        expected_index_metadata.search_settings
+    );
+    assert_eq!(
```

Comment:
FYI, after rebasing, this line will become:
```rust
assert_eq!(
        index_metadata.sources,
        expected_index_metadata.sources
    );
```

Reply:
fixed